mobile/ios/auth/ArcadiaAuthenticationSession.m

```diff
@@ -0,0 +1,68 @@
+//
+// Created by Tsimokha, Yaroslav on 2018-12-13.
+// Copyright (c) 2018 JSC Arcadia Inc. All rights reserved.
+//
+
+#import <AuthenticationServices/AuthenticationServices.h>
+#import "ArcadiaAuthenticationSession.h"
+
+NSString *const CANCELLATION_SUBCODE = @"error_subcode=cancel";
+
+//============================================================================
+@implementation ArcadiaAuthenticationSession
+
+//----------------------------------------------------------------------------
+- (dispatch_queue_t)methodQueue {
+    return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_MODULE()
+
+//----------------------------------------------------------------------------
+RCT_REMAP_METHOD(getSafariData, address:(NSString *)address
+            callbackURL:(NSString *)callbackURL
+            resolver:(RCTPromiseResolveBlock)resolve
+            rejecter:(RCTPromiseRejectBlock)reject)
+{
+    if (@available(iOS 12.0, *)) {
+
+        NSURL *siteURL = [NSURL URLWithString:address];
+        self.authSession = [[ASWebAuthenticationSession alloc] initWithURL:siteURL
+                                                         callbackURLScheme:callbackURL
+                                                         completionHandler:^(NSURL *cbURL, NSError *error) {
+
+                                                             if (error != nil) {
+                                                                 NSLog(@"%@", [error localizedDescription]);
+
+                                                                 reject([@(error.code) stringValue], [error localizedDescription], error);
+                                                                 return;
+                                                             }
+
+                                                             if ([self isCancellationUrl:cbURL]) {
+                                                                 reject([@(ASWebAuthenticationSessionErrorCodeCanceledLogin) stringValue], @"Back button pressed", nil);
```

Comment:
I believe it's better to use our app specific error code there.

Reply:
73459b2fc09a778ee19b01c4011e1d63f4717ced